src/features/user/TreeMapper/components/Map.tsx

```diff
@@ -0,0 +1,357 @@
+  selectedLocation,
+  setselectedLocation,
+}: Props): ReactElement {
+  const router = useRouter();
+  const { useTranslation } = i18next;
+  const { i18n, t } = useTranslation('me');
+
+  const {isMobile} = React.useContext(ProjectPropsContext);
+
+  const defaultMapCenter = [-28.5, 36.96];
+  const defaultZoom = 1.4;
+  const [viewport, setViewPort] = React.useState({
+    width: Number('100%'),
+    height: Number('100%'),
+    latitude: defaultMapCenter[0],
+    longitude: defaultMapCenter[1],
+    zoom: defaultZoom,
+  });
+  const [satellite, setSatellite] = React.useState(false);
+  
+  const [geoJson, setGeoJson] = React.useState();
+  const [plIds, setPlIds] = React.useState(null);
+  const [imagePopup, setImagePopup] = React.useState(null);
+  let timer: NodeJS.Timeout;
+
+  const [style, setStyle] = React.useState({
+    version: 8,
+    sources: {},
+    layers: [],
+  });
+
+  const getPlTreeCount = (pl: any) => {
+    let count = 0;
+    if (pl && pl.plantedSpecies) {
+      for (const key in pl.plantedSpecies) {
+        if (Object.prototype.hasOwnProperty.call(pl.plantedSpecies, key)) {
+          const element = pl.plantedSpecies[key];
+          count += element.treeCount;
+        }
+      }
+      return count;
+    } else {
+      return 0;
+    }
+  };
+
+  const getPlArea = (pl: any) => {
+    if (pl && pl.type === 'multi') {
+      const area = turf.area(pl.geometry);
+      return area / 10000;
+    } else {
+      return 0;
+    }
+  };
+
+  const getPolygonColor = (pl: any) => {
+    const treeCount = getPlTreeCount(pl);
+    const plantationArea = getPlArea(pl);
+    const density = treeCount / plantationArea;
+    if (density > 2500) {
+      return 0.5;
+    } else if (density > 2000) {
+      return 0.4;
+    } else if (density > 1600) {
+      return 0.3;
+    } else if (density > 1000) {
+      return 0.2;
+    } else {
+      return 0.1;
+    }
+  };
+
+  const getDateDiff = (pl: any) => {
+    const today = new Date();
+    const plantationDate = new Date(pl.plantDate?.substr(0,10));
+    const differenceInTime = today.getTime() - plantationDate.getTime();
+    const differenceInDays = differenceInTime / (1000 * 3600 * 24);
+    if (differenceInDays < 1) {
+      return t('today');
+    } else if (differenceInDays < 2) {
+      return t('yesterday');
+    } else if (differenceInDays < 30) {
+      return t('daysAgo', {
+        days: localizedAbbreviatedNumber(i18n.language, differenceInDays, 0),
+      });
+    } else {
+      return null;
+    }
+  };
+
+  const zoomToLocation = (geometry: any) => {
+    try {
+  const bbox = turf.bbox(geometry);
+
+  const { longitude, latitude, zoom } = new WebMercatorViewport(
+    viewport
+  ).fitBounds(
+    [
+      [bbox[0], bbox[1]],
+      [bbox[2], bbox[3]],
+    ],
+    {
+      padding: {
+        top: isMobile ? 50: 100,
+        bottom: isMobile ? 200 : 100,
+        left: isMobile ? 50 : 100,
+        right: isMobile ? 50 : 100,
+      },
+    }
+  );
```

Comment:
This code has the risk of crashes in `WebMercatorViewport` as we have tried to fix them at https://github.com/Plant-for-the-Planet-org/planet-webapp/blob/fb9a7002e3950045d5db57794440df4d27a5b1a0/src/utils/maps/zoomToProjectSite.ts#L13-L25

Reply:
I also have not added these checks at https://github.com/Plant-for-the-Planet-org/planet-webapp/blob/fb9a7002e3950045d5db57794440df4d27a5b1a0/src/utils/maps/plantLocations.ts#L14

Reply:
I have added try-catch, the values are NaN sometimes.

Reply:
@sarveshpro I could not see any commit from you regarding this change?